monadic-html/src/main/scala/mhtml/config.scala

```diff
@@ -0,0 +1,228 @@
+package mhtml
+
+import java.util.Arrays.binarySearch
+
+case class Config(
+  transformEntityRef:   String => String,
+  transformAtom:           Any => String,
+  transformAttributeValue: Any => String,
+  checkElement:         String => Unit,
+  checkEventKey:        String => Unit,
+  checkAttributeKey:    String => Unit
+)
+
+object Config {
+  import ConfigHelper._
+
+  /** Empty configuration with minimal impact on binary size. */
+  val empty = Config(
+    transformEntityRef      = identity,
+    transformAtom           = _.toString,
+    transformAttributeValue = _.toString,
+    checkElement            = _ => (),
+    checkEventKey           = _ => (),
+    checkAttributeKey       = _ => ()
+  )
+
+  /** Default production configuration; takes care of EntityRef conversion. */
+  val entityRefOnly = empty.copy(transformEntityRef = silentEntityRef)
+
+  def verbose(warn: String => Unit) = Config(
+    transformEntityRef      = warnEntityRef(warn),
+    transformAtom           = warnAny2String(warn),
+    transformAttributeValue = warnAny2String(warn),
+    checkElement            = checkElement(warn),
+    checkEventKey           = checkEvent(warn),
+    checkAttributeKey       = checkAttribute(warn)
+  )
+
+  /** Default development configuration; warns about unknown
+    * elements/attributes/entityRefs and implicit Any to String conversions. */
+  val default = verbose(s => println(s"[mhtml] Warning: $s"))
```

Comment:
Might be good to also provide a `failFast` config that throws errors instead of printing them.